includes/discovery/functions.inc.php

```diff
@@ -1363,7 +1363,7 @@ function find_device_id($name = '', $ip = '', $mac_address = '')
     $where = array();
     $params = array();
 
-    if ($name && is_valid_hostname($name)) {
```

Comment:
You should at least check that it isn't an IP or some other garbage input.

Reply:
The `hostname` field can contain an IP address if the user added a device by specifying an IP address instead of a host name. In this case, the `ip` field in the devices table will be null. The `sysName` field is defined by the device and there can be any value. In my case, sysName contains the symbol _, which is not permitted by the RFC, but is accepted by most operating systems (the person who assigned the names did not know such subtleties). When announcing itself through LLDP, the device sends its sysName, including the _ character, and the LLDP-discovery module cannot find my device in the database, although it is there.
In addition, the `find_device_id` function searches for, but does not change, data in the database. In my opinion, the search function should not present any complex requirements for the search name. If the object in the database has any non-standard name, you still need to be able to find it and work with it. Also, the search takes place on the sysName field, which the user sets at his own discretion.

Reply:
worried about false positives?  sysName is not unique you could have multiple devices with the same sysName.

Reply:
It’s appropriate. Although the current implementation of non-unique sysName does not take into account. I suggest the following sequence:
1. verification by hostname - the original version, with verification through is_valid_hostname
2. ip check - original version
3. verification by sysName - verification without is_valid_hostname, with verification of the number of matches. If more than one - do not return anything.
4. mac check - original version

Reply:
I would put the mac check before the sysName.

Reply:
Done.

Reply:
@murrant ping